src/evaluators/bc/PHAL_SDirichlet_Def.hpp

```diff
@@ -86,6 +86,26 @@ SDirichlet<PHAL::AlbanyTraits::Jacobian, Traits>::SDirichlet(
 {
 }
 
+
+template <typename Traits>
+void
+SDirichlet<PHAL::AlbanyTraits::Jacobian, Traits>::preEvaluate(
+    typename Traits::EvalData dirichlet_workset)
+{
+  if(Teuchos::nonnull(dirichlet_workset.f)) {
```

Comment:
How come tests passed without this pre-evaluate? Is this really necessary?

Reply:
I can imagine that during the optimization loop one could ask to compute the Jacobian and residual at the same time. It does not seem to happen now.

Reply:
I see, thanks.